test/tracerEq/test_h-advection_mes_2d.py

```diff
@@ -0,0 +1,204 @@
+"""
+Testing 2D horizontal advection of tracers
+
+Tuomas Karna, edited by Athanasios Angeloudis
+"""
+from thetis import *
+import numpy
+from scipy import stats
+import pytest
+
+
+def run(refinement, **model_options):
+    print_output('--- running refinement {:}'.format(refinement))
+
+    # domain dimensions - channel in x-direction
+    lx = 15.0e3
+    ly = 6.0e3/refinement
+    area = lx*ly
+    depth = 40.0
+    u = 1.0
+
+    # mesh
+    nx = 6*refinement + 1
+    ny = 1  # constant -- channel
+    mesh2d = RectangleMesh(nx, ny, lx, ly)
+
+    # simulation run time
+    t_end = 3000.0
+    # initial time
+    t_init = 0.0
+    t_export = (t_end - t_init)/8.0
+
+    # outputs
+    outputdir = 'outputs'
+
+    # bathymetry
+    P1_2d = FunctionSpace(mesh2d, 'CG', 1)
+    bathymetry_2d = Function(P1_2d, name='Bathymetry')
+    bathymetry_2d.assign(depth)
+
+    solverobj = solver2d.FlowSolver2d(mesh2d, bathymetry_2d)
+    options = solverobj.options
+    options.use_nonlinear_equations = False
+    options.use_lax_friedrichs_velocity = True
+    options.lax_friedrichs_velocity_scaling_factor = Constant(1.0)
+    options.use_lax_friedrichs_tracer = False
+    options.horizontal_velocity_scale = Constant(abs(u))
+    options.no_exports = True
+    options.output_directory = outputdir
+    options.simulation_end_time = t_end
+    options.simulation_export_time = t_export
+    options.solve_tracer = True
+    options.use_limiter_for_tracers = True
+    options.fields_to_export = ['tracer_2d']
+    options.update(model_options)
+    uv_expr = as_vector((u, 0))
+    bnd_salt_2d = {'value': Constant(0.0), 'uv': uv_expr}
+    bnd_uv_2d = {'uv': uv_expr}
+    solverobj.bnd_functions['tracer'] = {
+        1: bnd_salt_2d,
+        2: bnd_salt_2d,
+    }
+    solverobj.bnd_functions['momentum'] = {
+        1: bnd_uv_2d,
+        2: bnd_uv_2d,
+    }
+
+    solverobj.create_equations()
+
+    t = t_init  # simulation time
+
+    x0 = 0.3*lx
+    sigma = 1600.
+    xy = SpatialCoordinate(solverobj.mesh2d)
+    t_const = Constant(t)
+    ana_tracer_expr = exp(-(xy[0] - x0 - u*t_const)**2/sigma**2)
+    tracer_ana = Function(solverobj.function_spaces.H_2d, name='tracer analytical')
+
+    p1dg_ho = FunctionSpace(solverobj.mesh2d, 'DG', options.polynomial_degree + 2,
+                            vfamily='DG', vdegree=options.polynomial_degree + 2)
+    tracer_ana_ho = Function(p1dg_ho, name='tracer analytical')
```

Comment:
I guess my point on the high order projection was that `p1dg_ho` and `tracer_ana_ho` are no longer needed. You can just compute the error as `errornorm(tracer_ana_ufl_expr, tracer_sol)`. I think `/test/pressure_grad` has some examples (`degree_rise` option is also obsolete).

Reply:
Oh now I see, yes I just realised that these do not make a difference! Updating it now